img2dataset/downloader.py

```diff
@@ -348,26 +401,63 @@ def download_one_file(url_list):
             timeout=timeout,
         )
 
+        # Start a W&B run
+        run = wandb.init(
+            project="img2dataset", config=config_parameters, anonymous="allow"
+        )
+
+        total_speed_logger = SpeedLogger("total")
+        # capture error messages
+        total_status_dict = {}
+
         total_total = 0
         total_success = 0
         total_failed_to_download = 0
         total_failed_to_resize = 0
+
         with Pool(processes_count, maxtasksperchild=5) as process_pool:
-            for total, successes, failed_to_download, failed_to_resize in tqdm(
+            for (
+                total,
+                successes,
+                failed_to_download,
+                failed_to_resize,
+                speed_logger,
+                status_dict,
+            ) in tqdm(
                 process_pool.imap_unordered(downloader, sharded_images_to_dl),
                 total=len(sharded_images_to_dl),
+                file=sys.stdout,
             ):
                 total_total += total
                 total_success += successes
                 total_failed_to_download += failed_to_download
                 total_failed_to_resize += failed_to_resize
-                message = f"success={1.0*total_success/total_total:.2f} "
-                message += (
-                    f"failed download={1.0*total_failed_to_download/total_total:.2f} "
+
+                speed_logger(
+                    count=total,
+                    success=successes,
+                    failed_to_download=failed_to_download,
+                    failed_to_resize=failed_to_resize,
+                )
+                total_speed_logger(
+                    count=total_total,
+                    success=total_success,
+                    failed_to_download=total_failed_to_download,
+                    failed_to_resize=total_failed_to_resize,
                 )
-                message += f"failed resize={1.0*total_failed_to_resize/total_total:.2f}"
-                print(message + "\n", sep=" ", end="", flush=True)
-                pass
+
+                # update status table
+                for k, v in status_dict.items():
+                    total_status_dict[k] = total_status_dict.get(k, 0) + v
```

Comment:
I will need to double check this doesn't grow out of control by doing a larger run